gcc/config/sol2.h

```diff
@@ -197,8 +197,14 @@ along with GCC; see the file COPYING3.  If not see
 #undef STARTFILE_ARCH_SPEC
 #define STARTFILE_ARCH_SPEC \
   "%{!shared:%{!symbolic: \
-     %{ansi|std=c*|std=iso9899\\:199409:values-Xc.o%s; :values-Xa.o%s} \
-     %{std=c90|std=gnu90:values-xpg4.o%s; :values-xpg6.o%s}}}"
+     %{ \
+	std=c89|std=c90|std=gnu89|std=gnu90:values-Xa.o%s; \
+	ansi|std=c*|std=iso9899\\:199409:values-Xc.o%s; \
+	:values-Xa.o%s} \
+     %{ \
+	std=c89|std=c90|std=gnu89|std=gnu90:; \
```

Comment:
Does this not leave us failing to select values-xpg4 at all now?

Reply:
Oh. Right!  I remember this now, I think.

This is the thing where there's no compiler-y flag that actually implies xpg4, because that's entirely orthogonal to the C standard (whereas xpg6 and c99 are intimately connected), right?

Reply:
Strictly speaking xpg6 implies c99, but the way we've done everything per `lib/libc/inc/xpg6.h` basically says that c99 behavior is values-xpg6.o. And I guess we'd want to make sure that we're in c11 mode as well.

xpg4 requires c89, but I think that your general note is correct that at least as libc is concerned, xpg4 is orthogonal to the C standard.

Reply:
A somewhat naïve comparison between gcc-7-il and gcc-8 with this patch is below:
The results are consistent apart from gnu99 and gnu11 without `-ansi`. Do either of you have any insights on that?

@tsoome - was this one of the sources of your problems with gcc9 and _lib_version?

| flags | gcc7 | gcc8 |
| -- | -- | -- |
| c89 32  | Xa  | Xa  |
| c89 32 -ansi | Xa  | Xa  |
| c89 64  | Xa  | Xa  |
| c89 64 -ansi | Xa  | Xa  |
| c90 32  | Xa  | Xa  |
| c90 32 -ansi | Xa  | Xa  |
| c90 64  | Xa  | Xa  |
| c90 64 -ansi | Xa  | Xa  |
| c99 32  | Xc xpg6  | Xc xpg6  |
| c99 32 -ansi | Xc xpg6  | Xc xpg6  |
| c99 64  | Xc xpg6  | Xc xpg6  |
| c99 64 -ansi | Xc xpg6  | Xc xpg6  |
| c11 32  | Xc xpg6  | Xc xpg6  |
| c11 32 -ansi | Xc xpg6  | Xc xpg6  |
| c11 64  | Xc xpg6  | Xc xpg6  |
| c11 64 -ansi | Xc xpg6  | Xc xpg6  |
| gnu89 32  | Xa  | Xa  |
| gnu89 32 -ansi | Xa  | Xa  |
| gnu89 64  | Xa  | Xa  |
| gnu89 64 -ansi | Xa  | Xa  |
| gnu90 32  | Xa  | Xa  |
| gnu90 32 -ansi | Xa  | Xa  |
| gnu90 64  | Xa  | Xa  |
| gnu90 64 -ansi | Xa  | Xa  |
| gnu99 32  | Xc xpg6  | Xa xpg6  |
| gnu99 32 -ansi | Xc xpg6  | Xc xpg6  |
| gnu99 64  | Xc xpg6  | Xa xpg6  |
| gnu99 64 -ansi | Xc xpg6  | Xc xpg6  |
| gnu11 32  | Xc xpg6  | Xa xpg6  |
| gnu11 32 -ansi | Xc xpg6  | Xc xpg6  |
| gnu11 64  | Xc xpg6  | Xa xpg6  |
| gnu11 64 -ansi | Xc xpg6  | Xc xpg6  |